### PR TITLE
Build sandbox website with svelte

### PR DESCRIPTION
**Pull request by johannchopin**
The goal is to develop the website https://jakubpawlowicz.github.io/clean-css/ in a more modern way by using [svelte](https://svelte.dev/). It will simplfy the development of future features like #1164.

### CODE REVIEW

**Comment by johannchopin** [comment]
@jakubpawlowicz Finally finished the first try of the new website with svelte. It simple use some decoupled components and a store to manage the global settings wanted by the user. I also added a GitHub Action to deploy it on the `gh-pages` branch. You can check it live: https://johannchopin.github.io/clean-css/

**Comment by jakubpawlowicz** [comment]
Thanks @johannchopin - you are fast! I'll have a closer look on Wednesday as I'm not in front of my computer today and tomorrow.

**Comment by jakubpawlowicz** [comment]
Very cool @johannchopin I like what you did here.

Couple questions:
1) Should I review it as it is or do you plan to change / add stuff?
2) Should we have a Worker instance for running the optimizations? I kinda like it but I'm open to ideas.

Thanks!
Jakub

**Comment by jakubpawlowicz** [comment]
Oh, and you may want to change jshint config to respect ES6 syntax, see: 
https://github.com/jakubpawlowicz/clean-css/pull/1165/checks?check_run_id=2894299597

**Comment by johannchopin** [comment]
JsHint sadly doesn't seems to handle svelte like it's possible with eslint. Therefore I guess it's a good idea to use eslint for the scope of the website and ignore the `/website` folder from jshint.

> Should we have a Worker instance for running the optimizations? I kinda like it but I'm open to ideas.

Workers are really great but could lead to unnecessary complexity. This website works synchronously (drop the file and you need to wait for the result) so I don't think that we need to run the optimizations in the backgroung. To keep it simple I guess it's better to call the `CleanCSS` instance simply when it's needed.

If it's 👌 for you it would be ready for a review 🚀

**Comment by jakubpawlowicz** [comment]
Cool, thanks for updates. The only reason I'm asking about Worker is that if you drop a heavy CSS file, let's say 1MB+, the UI will be stuck for some time, as optimizations happen. Let me think about it a little more.

I will review the code none the less as it should be feature complete now.

**Comment by johannchopin** [comment]
@jakubpawlowicz From a performance point of view this should be fine. I was able to minify all `.css` file from bootstrap's dist folder without noticing anything 👍

File: website/src/components/FileDropzone.svelte [hunk 1]
```diff
@@ -0,0 +1,78 @@
+<script>
+  import Dropzone from 'svelte-file-dropzone'
+  import * as CleanCSS from 'clean-css'
+  import { options } from '../stores'
+
+import Legend from './Legend.svelte'
```
Comment (jakubpawlowicz) [inline website/src/components/FileDropzone.svelte:6]: Wrong indent?

File: website/src/components/FileDropzone.svelte [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      reader.readAsText(file)
+    })
+  }
+</script>
+
+<div class="d-flex flex-column justify-center">
+  <Dropzone 
+    on:dropaccepted={handleFilesSelect}
+    on:droprejected={() => { isDroppedFileErrored = true }}
+    accept={['text/css']}
```
Comment (jakubpawlowicz) [inline website/src/components/FileDropzone.svelte:50]: Cool :+1:

File: website/src/components/LoadedFile.svelte [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    const blob = new Blob([content], { type: 'text/css;charset=utf-8' })
+    saveAs(blob, name)
+  }
+</script>
+
+<li class="list-group list-group-item d-flex flex-row align-items-center p-1">
+  <span class="fw-bold">{name}</span>
+  <span class="fw-lighter fst-italic"> - saved {sizeSaved / 1000} kB</span>
+  <div class="d-flex justify-content-end flex-grow-1">
+    <button class="btn btn-link p-1 ms-4" on:click={saveToClipboard}>copy to clipboard</button>
```
Comment (jakubpawlowicz) [inline website/src/components/LoadedFile.svelte:23]: Old version does a very simple text change to "Done!" when clicked. Can we do it too?

File: website/src/main.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import App from './App.svelte'
+
+import 'bootstrap/dist/css/bootstrap.min.css'
+
+const app = new App({
+  target: document.body,
+  props: {
+    name: 'world'
```
Comment (jakubpawlowicz) [inline website/src/main.js:8]: Should it be around?

File: website/public/index.html [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+	<meta name='viewport' content='width=device-width,initial-scale=1'>
+
+	<title>clean-css: fast and efficient CSS optimizer</title>
+
+	<link rel='icon' type='image/png' href='favicon.png'>
+	<link rel='stylesheet' href='global.css'>
+	<link rel='stylesheet' href='build/bundle.css'>
+
+	<script defer src='build/bundle.js'></script>
+</head>
```
Comment (jakubpawlowicz) [inline website/public/index.html:14]: Can we have a consistent 2 space indent here?

File: website/src/App.svelte [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+	import FileDropzone from './components/FileDropzone.svelte'
+	import Settings from './components/Settings.svelte'
+</script>
+
+<Header />
+<main class="d-flex flex-column justify-content-center align-items-center px-2 px-sm-5">
+	<Settings />
+	<FileDropzone />
+	<Footer />
+</main>
```
Comment (jakubpawlowicz) [inline website/src/App.svelte:13]: And here as well re 2 space indent?

**Review by jakubpawlowicz — COMMENTED** [review]
Overall :100: just couple minor comments! Well done @johannchopin !

File: website/src/main.js (hunk 4 above)
Comment (johannchopin) [reply website/src/main.js]: Nope this props come with the initial boilerplate I removed it 👍

File: website/src/components/LoadedFile.svelte (hunk 3 above)
Comment (johannchopin) [reply website/src/components/LoadedFile.svelte]: Sure and done 👍

**Comment by jakubpawlowicz** [comment]
One last thing - build actions fail because of missing 'eslint' - is that because `npm run check` runs website linter too now?

**Comment by jakubpawlowicz** [comment]
I think we need to add 'eslint' as a devDependency to main `package.json` to make it work without `npm install` inside `website` dir.

**Comment by johannchopin** [comment]
@jakubpawlowicz Issue is that I used `yarn` to run it so I fixed that and also I needed to install the dependencies in `./website` before running the `check` command 😅

**Comment by jakubpawlowicz** [comment]
Cool, looks like we are green. I'm going to squash all commits into one - any objections?

**Comment by johannchopin** [comment]
@jakubpawlowicz Nope just go ahead 🚀

**Comment by johannchopin** [comment]
@jakubpawlowicz Have you set in GitHub page to look at the `gh-pages` branch?

**Comment by jakubpawlowicz** [comment]
I was looking for it as you wrote! Now it's set up.

**Comment by jakubpawlowicz** [comment]
Looks like it works, :100: @johannchopin

Please feel free to implement #1164 when you please.

**Pull request merged**